packages/node/src/policy-gen/load-for-policy.js

```diff
@@ -0,0 +1,463 @@
+import { WorkerPool } from '../worker-pool.js'
+
+/**
+ * @import {LoadAndGeneratePolicyOptions, LoadCompartmentMapResult, InspectMessage, PoliciesMessage, ErrorMessage, StructuredViolationsResult} from '../internal.js'
+ * @import {CanonicalName, Language, LocalModuleSource} from '@endo/compartment-mapper'
+ * @import {BuiltinPolicy, GlobalPolicy, GlobalPolicyValue, LavaMoatPolicy, PackagePolicy} from '@lavamoat/types'
+ * @import {MergedLavaMoatPolicy, FileUrlString, SourceType} from '../types.js'
+ * @import {Loggerr} from 'loggerr'
+ */
+
+const inspectorPath = fileURLToPath(new URL('./inspector.js', import.meta.url))
+
+const { entries, keys } = Object
+
+/**
+ * The nitty-gritty of building a policy from a given entrypoint.
+ *
+ * Loads the compartment map and subscribes to hooks to inspect modules for
+ * globals, builtins, and SES compatibility violations, then builds a policy
+ * from it. This bypasses `lavamoat-core`'s inspector entirely.
+ *
+ * @param {string | URL} entrypointPath
+ * @param {LoadAndGeneratePolicyOptions} options
+ * @returns {Promise<LoadCompartmentMapResult>}
+ * @internal
+ */
+export const loadAndGeneratePolicy = async (
+  entrypointPath,
+  {
+    readPowers = defaultReadPowers,
+    policyOverride,
+    trustRoot = DEFAULT_TRUST_ROOT_COMPARTMENT,
+    log = defaultLog,
+    dev,
+    ...options
+  } = {}
+) => {
+  const {
+    packageJsonMap,
+    packageCompartmentMap,
+    unknownCanonicalNames,
+    knownCanonicalNames,
+  } = await makeNodeCompartmentMap(entrypointPath, {
+    readPowers,
+    dev,
+    log,
+    trustRoot,
+  })
+
+  /**
+   * In-flight requests to inspect modules
+   *
+   * @type {Set<Promise<FileUrlString>>}
+   */
+  const pendingInspections = new Set()
+
+  /**
+   * All modules to inspect. This will grow as our `moduleSource` hooks are
+   * called.
+   *
+   * @type {Set<FileUrlString>}
+   */
+  const modulesToInspect = new Set()
+
+  /**
+   * List of modules that have been inspected. This will grow as our worker pool
+   * messages are processed.
+   *
+   * @type {Set<FileUrlString>}
+   */
+  const inspectedModules = new Set()
+
+  /**
+   * A worker pool for inspecting modules.
+   *
+   * @type {WorkerPool<InspectMessage, PoliciesMessage | ErrorMessage>}
+   */
+  const workerPool = new WorkerPool(inspectorPath)
+
+  /**
+   * Stores global policies for each package.
+   *
+   * Updated as we receive messages from the worker pool.
+   *
+   * @type {Map<CanonicalName, GlobalPolicy>}
+   */
+  const globalsForPackage = new Map()
+
+  /**
+   * Stores builtin policies for each package.
+   *
+   * Updated as we receive messages from the worker pool.
+   *
+   * @type {Map<CanonicalName, BuiltinPolicy>}
+   */
+  const builtinsForPackage = new Map()
+
+  /**
+   * Stores package policies for each package.
+   *
+   * Updated as we receive messages from the worker pool.
+   *
+   * @type {Map<CanonicalName, PackagePolicy>}
+   */
+  const packagesForPackage = new Map()
```

Comment:
(could) rename to `connectionsForPackage` ?

Reply:
`packagePoliciesMap` seemed more appropriate to me, so there it is.